'Ambiguous reference' error when building with SPM 4β3

Builds using Swift Package Manager 4 Beta 3 fail with `ambiguous reference to member 'add'` error in [Signal.swift:1067](https://github.com/ReactiveCocoa/ReactiveSwift/blob/master/Sources/Signal.swift#L1067).

---

#### Package.swift

```swift
.package(url: "https://github.com/ReactiveCocoa/ReactiveSwift", from: "2.0.0-rc.3")
```

#### swift build -Xswiftc -suppress-warnings

```none
.build/checkouts/ReactiveSwift-3050645160503825854/Sources/Signal.swift:1067:21: error: ambiguous reference to member 'add'
                        _ = disposed.map(disposable.add)
                                         ^~~~~~~~~~
.build/checkouts/ReactiveSwift-3050645160503825854/Sources/Disposable.swift:164:14: note: found this candidate
        public func add(_ disposable: Disposable?) -> Disposable? {
                    ^
.build/checkouts/ReactiveSwift-3050645160503825854/Sources/Disposable.swift:191:14: note: found this candidate
        public func add(_ action: @escaping () -> Void) -> Disposable? {
                    ^

error: terminated(1): /Applications/Xcode-9.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-build-tool -f .build/debug.yaml
```

#### swift -version

```none
Apple Swift version 4.0 (swiftlang-900.0.52 clang-900.0.29)
Target: x86_64-apple-macosx10.9
```

#### xcodebuild -version

```none
Xcode 9.0
Build version 9M174d
```

--- a/Sources/Signal.swift
+++ b/Sources/Signal.swift
@@ -1064,7 +1064,9 @@ extension Signal {
 		return Signal { observer in
 			let disposable = CompositeDisposable()
 
-			_ = disposed.map(disposable.add)
+			if let action = disposed {
+				disposable.add(action)
+			}
 
 			disposable += self.observe { receivedEvent in
 				event?(receivedEvent)

